src/components/graph/menus/node-editor.js

```diff
@@ -5,41 +5,101 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Paper } from '@material-ui/core';
 import PropTypes from 'prop-types';
 import makeStyles from '@material-ui/core/styles/makeStyles';
 import NetworkModificationNodeEditor from './network-modification-node-editor';
 import ModelNodeEditor from './model-node-editor';
+import {
+    fetchNetworkModificationTreeNode,
+    updateTreeNode,
+} from '../../../utils/rest-api';
+import {
+    displayErrorMessageWithSnackbar,
+    useIntlRef,
+} from '../../../utils/messages';
+import { EditableTitle } from './editable-title';
+import { useSnackbar } from 'notistack';
+import { useSelector } from 'react-redux';
 
 const useStyles = makeStyles((theme) => ({
     root: {
         flexGrow: 1,
     },
     paper: {
-        height: '100%',
-        padding: theme.spacing(2),
+        height: 'max-content',
+        display: 'flex',
+        flexDirection: 'column',
+        flexGrow: 1,
         elevation: 3,
     },
 }));
 
-const NodeEditor = ({ selectedNode, handleNodeModified }) => {
+const NodeEditor = ({ onClose, ...props }) => {
     const classes = useStyles();
+    const intlRef = useIntlRef();
+    const { enqueueSnackbar } = useSnackbar();
+    const [selectedNode, setSelectedNode] = useState();
+
+    const selectedNodeUuid = useSelector((state) => state.selectedTreeNode);
+
+    useEffect(() => {
+        if (!selectedNodeUuid) return;
+        fetchNetworkModificationTreeNode(selectedNodeUuid)
+            .then((res) => setSelectedNode(res))
+            .catch((err) =>
+                displayErrorMessageWithSnackbar({
+                    errorMessage: err.message,
+                    enqueueSnackbar,
+                })
+            );
+    }, [setSelectedNode, enqueueSnackbar, selectedNodeUuid]);
+
+    const changeNodeName = (newName) => {
+        updateTreeNode({
+            id: selectedNode?.id,
+            type: selectedNode?.type,
+            name: newName,
+        }).catch((errorMessage) => {
+            displayErrorMessageWithSnackbar({
+                errorMessage: errorMessage,
+                enqueueSnackbar: enqueueSnackbar,
+                headerMessage: {
+                    headerMessageId: 'NodeUpdateError',
+                    intlRef: intlRef,
+                },
+            });
+        });
+    };
 
     return (
-        <Paper className={classes.paper}>
-            {selectedNode.type === 'NETWORK_MODIFICATION' && (
-                <NetworkModificationNodeEditor selectedNode={selectedNode} />
-            )}
-            {selectedNode.type === 'MODEL' && (
-                <ModelNodeEditor selectedNode={selectedNode} />
+        <div {...props}>
```

Comment:
Why do you pass props to div here ? is it not a bit dangerous or too flexible ?

Reply:
changed, but I don't think flexibility is not a bad thing.